### PR TITLE
Hide PreInstaller implementation structs, improve tests

### PR DESCRIPTION
**Pull request by mjakobczyk**
**Description**

Changes proposed in this pull request:

- remove `output` struct from API (only used as a part of internal methods)
- unexport internal structs and constructor method
- improve tests by additional checks

**Related issue(s)**

See: https://github.com/kyma-project/hydroform/issues/265

### CODE REVIEW

File: parallel-install/pkg/deployment/preinstaller.go [hunk 1]
```diff
@@ -112,8 +113,8 @@ func NewPreInstaller(cfg Config) (*preInstaller, error) {
 }
 
 // InstallCRDs on a k8s cluster.
-// Returns Output containing results of installation.
-func (i *preInstaller) InstallCRDs() (Output, error) {
+// Returns output containing results of installation.
```
Comment (clebs) [inline parallel-install/pkg/deployment/preinstaller.go:116]: This no longer returns an output right?

File: parallel-install/pkg/deployment/preinstaller.go [hunk 2]
```diff
@@ -123,16 +124,16 @@ func (i *preInstaller) InstallCRDs() (Output, error) {
...
 	}
 
-	return output, nil
+	return nil
 }
 
 // CreateNamespaces in a k8s cluster.
-// Returns Output containing results of installation.
-func (i *preInstaller) CreateNamespaces() (Output, error) {
+// Returns output containing results of installation.
```
Comment (clebs) [inline parallel-install/pkg/deployment/preinstaller.go:135]: This no longer returns an output right?

**Review by clebs — APPROVED** [review]

**Comment by piotrkpc** [comment]
looks good

**Review by strekm — APPROVED** [review]
/meow

**Pull request merged**